var/spack/repos/builtin/packages/hypre/package.py

```diff
@@ -36,6 +36,7 @@ class Hypre(Package):
     url      = "http://computation.llnl.gov/project/linear_solvers/download/hypre-2.10.0b.tar.gz"
 
     version('develop', git='https://github.com/LLNL/hypre', tag='master')
+    version('2.12.1', git='https://github.com/LLNL/hypre', tag='v2.12.1', preferred='True')
```

Comment:
I don’t think you need to mark it as preferred. The latest stable release is always preferred

Reply:
Why not:
```
version('2.12.1', 'd507943a1a3ce5681c3308e2f3a6dd34', url='https://github.com/LLNL/hypre/archive/v2.11.2.tar.gz')
```

Reply:
actually
```
version('2.12.1', 'd507943a1a3ce5681c3308e2f3a6dd34')
```